core/src/test/java/software/amazon/awssdk/util/VersionInfoTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.Test;
+
+public final class VersionInfoTest {
+
+    @Test
+    public void versionIsTheSameAsMavenProject() throws Exception {
+        assertThat(VersionInfo.SDK_VERSION).isEqualTo(getSdkVersionFromPom());
+    }
+
+    private String getSdkVersionFromPom() throws URISyntaxException, IOException {
```

Comment:
@sarxos happy to go take another look, but this generates an actual class that (I assume) won't be compiled if it's considered to be a resource by maven, it'll just get copied into the target/classes directory as is.